src/Mvc/Mvc.Core/src/ModelBinding/Validation/DefaultComplexObjectValidationStrategy.cs

```diff
@@ -107,7 +107,7 @@ public bool MoveNext()
             else
             {
                 var property = _properties[_index - _parameters.Count];
-                var propertyName = property.BinderModelName ?? property.PropertyName;
+                var propertyName = property.ValidationModelName ?? property.BinderModelName ?? property.PropertyName;
```

Comment:
What would happen if we had the new MetadataProviders implement `IBindingMetadataProvider` instead and change the `BinderModelName`? Would this have negative consequences in other areas? I don't want to create too many different kinds of names.

I saw that we're also now changing the `DisplayName` in the providers. Assuming we can't change the `BinderModelName`, why not use the `DisplayName` instead and forget adding a new `ValidationModelName` property? Could we always register the providers that update the `DisplayName` (regardless of options) but only use the `DisplayName` here if `MvcOptions.UseDisplayNameInValidationErrors` is set?

I like being able to reduce the new API a single option. I hope we can make the new API opt-out. Maybe it could even be reduced to an AppContextSwitch and avoid new API entirely.

Reply:
Use `BinderModelName` was my first try and it works, however, I saw that the property is used all around the source code during the model binding and I decided to create a new property but during my quick test I did not get any bad behavior in use it but for sure will be riskier.

Using the DisplayName could be a good thing and I know reduce the API could be better, but I prefer to use a new property in this case to make it clearer what we are doing.  But we can go through this scenario during the API review.